tests/app/test_message.py

```diff
@@ -3,51 +3,127 @@
 import pytest
 from datetime import datetime
 import json
+from app.configuration import Configuration
 from app.message import CommandMessage, ErrorMessage
-from app.exceptions import InvalidMessageError
+from app.exceptions import InvalidMessageError, UnknownCommandError
 
 
-def test_good_cmd_message():
-    json_obj = {"action": "somecoil", "value": True}
-    json_str = json.dumps(json_obj)
-    read_msg = CommandMessage.read(json_str)
-    assert "action" in read_msg
-    assert "value" in read_msg
+class TestCommandMessage:
+    def setup_method(self):
+        self.configuration = Configuration.from_file(
+            "tests/config/example_configuration.yaml"
+        )
 
-
-def test_bad_cmd_message_structure():
-    json_obj = {"action": "somecoil", "value": True}
-    for key in ("action", "value"):
-        json_obj["foo"] = json_obj[key]
-        del json_obj[key]
+    def test_good_cmd_message(self):
+        json_obj = {"action": "somecoil", "value": True}
         json_str = json.dumps(json_obj)
+        read_msg = CommandMessage.read(json_str)
+        assert "action" in read_msg
+        assert "value" in read_msg
+
+    def test_unknown_command(self):
+        with pytest.raises(UnknownCommandError) as ex:
+            CommandMessage("bad_register", 54, self.configuration)
+        assert "No coil or register found to match 'bad_register'" in str(ex)
+        with pytest.raises(UnknownCommandError) as ex:
+            CommandMessage("", False, self.configuration)
+        assert "No coil or register found" in str(ex)
+
+    def test_bad_cmd_message_structure(self):
+        json_obj = {"action": "somecoil", "value": True}
+        for key in ("action", "value"):
+            json_obj["foo"] = json_obj[key]
+            del json_obj[key]
+            json_str = json.dumps(json_obj)
+            with pytest.raises(InvalidMessageError) as ex:
+                _ = CommandMessage.read(json_str)
+            assert "Message is missing required components" in str(ex.value)
+            assert ex.type == InvalidMessageError
+            json_obj[key] = json_obj["foo"]
+
+    def test_bad_cmd_message_syntax(self):
+        json_str = "not a real json string"
         with pytest.raises(InvalidMessageError) as ex:
             _ = CommandMessage.read(json_str)
-        assert "Message is missing required components" in str(ex.value)
+        assert "Message is invalid JSON syntax" in str(ex.value)
         assert ex.type == InvalidMessageError
-        json_obj[key] = json_obj["foo"]
+
+    def test_bad_cmd_message_datatype(self):
+        msg = CommandMessage("evgBatteryModeCoil", True, self.configuration)
+        try:
+            msg.validate()
+        except InvalidMessageError:
+            assert False
+        msg = CommandMessage("evgBatteryModeCoil", "foo", self.configuration)
+        with pytest.raises(InvalidMessageError) as ex:
+            msg.validate()
+        assert "The coil value 'foo' is invalid" in str(ex.value)
```

Comment:
In general, I avoid assertions on exception messages, as those should be able to vary more freely. However, is not wrong. Is not hard to maintain this anyway